### ISSUE TITLE
Review question: Ignoring warnings

### ISSUE THREAD

**Issue by lachlangrose** [nialov/fractopo#26]
Hi,

In the documentation the recommendation is to ignore warnings. I think this is potentially a bad idea as the warnings are usually there for a reason and it could have unintended consequences for other libraries that may be used in the same workflow. Would it be possible to filter the specific warnings which you deem at unncessary?

**Comment by lachlangrose** [comment]
Following on from this point, I have seen there are a lot of warnings from parts of this code which would not be difficult to remove.

For example to add another dataframe to an existing dataframe:

`df.append(addition, ignore_index=True)` can be replaced with `df = df.concat([df,addition],ignore_index=True)`

to do the same thing multiple times for example in ` determine_intersects` you can build up a list of intersecting dataframes and then using a python list and then concat all of these lists together once the loop has been run. 

This should speed up the code and will also make it more future proof. Another warning `ShapelyDeprecationWarning: The array interface is deprecated and will no longer work in Shapely 2.0. Convert the '.coords' to a numpy array instead.
  arr = construct_1d_object_array_from_listlike(values)` would be worth fixing as well.

**Comment by nialov** [comment]
Thank you for your feedback. The warning ignoring has been made more explicit
in our documentation, i.e. the notebooks, to avoid any unintended consequences.
Additionally, we have implemented numerous code fixes, including all the ones
you mentioned in your post, in pull request #28.

Furthermore, we have implemented a more strict approach regarding warnings. The
expected errors are now better explicitly listed in `pyproject.toml`. This was
also the case previously but your issue report prompted a general debugging and
cleaning session where I hunted where warnings came from to assess if they were
mostly harmless or related to underlying dependencies in which they were added
to `pyproject.toml`.

The debugging turned into a general bug hunting session for me so do not be
worried about the amount of commits that prompted from this issue report :D. It
is good to check through these more minor warnings every now and then to keep
the code base "more future proof" as you also pointed out.
